src/main/java/no/spid/api/client/SpidApiClient.java

```diff
@@ -423,6 +423,12 @@ private boolean refreshToken(SpidOAuthToken token) {
         return true;
     }
 
+    private Optional<String> getUserId(OAuthJSONAccessTokenResponse response) {
+        JSONObject jsonBody = new JSONObject(response.getBody());
+        String optString = jsonBody.optString("user_id");
+        return optString.equals("false") || optString.equals("") ? Optional.<String>empty() : Optional.of(jsonBody.getString("user_id"));
```

Comment:
1) I wonder what is a case when value of "user_id" is "false".
2) Optional.<String>empty() could be changed to Optional.empty()

Reply:
Ad.1. SPiD's Mongo usage results. We just need to deal with that.

Reply:
user_id is false when token is a client token. And then it's boolean false, not string false though.

Reply:
I actually thought I could remove the <String> as well (as suggested by IntelliJ), but for some reason it does not compile. At least intellij gave me a warning, so maybe the problem is with IntelliJ.

Reply:
optString() will translate boolean false to the string "false".
https://stleary.github.io/JSON-java/

Reply:
ok